### PR TITLE
Replace Sasslint with Stylelint

### PR DESCRIPTION
**Pull request by amazingrando**
## What this PR does
This PR replaces sasslint with [Stylelint.io](https://stylelint.io/), a more robust and thorough linter in the style of ESlint (including the ability to disable styles per line or per file).

## How to review
- [x] Download this branch
- [x] Run `yarn`. (I ran into issues and installed with `npm i`. You may have to clear the yarn or npm cache. LMK if you experience the same.)
- [x] Run `npm start` and see no errors.
- [x] Visit a .scss file and remove the same after a semi-colon, e.g. `background: red` becomes `background:red`. Run `npm start` and see the error.

## Before merging
- [x] We should determine our base rules. Personally I endorse using `stylelint-config-standard` as found in this PR.
- [x] `package.json` should point to Emulsify-gulp after that repo has merged the `stylelint` branch. Currently it points to the stylelint branch.

### CODE REVIEW

File: .stylelintrc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "extends": "stylelint-config-standard"
+}
```
Comment (ccjjmartin) [inline .stylelintrc]: Missing new line at end of file

File: package.json [hunk 2]
```diff
@@ -3,11 +3,15 @@
   "description": "Project template for Drupal 8 themes",
   "version": "1.0.0",
   "devDependencies": {
-    "emulsify-gulp": "fourkitchens/emulsify-gulp",
+    "emulsify-gulp": "fourkitchens/emulsify-gulp#stylelint",
```
Comment (ccjjmartin) [inline package.json]: Does this need to be it's own project or is there a better way to handle this?

Comment (amazingrando) [inline package.json]: This is temporary for reviewing the work. Once it works we'll merge the Emulsify Gulp PR and update this package.json.

File: .stylelintrc (hunk 1 above)
Comment (amazingrando) [inline .stylelintrc]: Fixed.

**Comment by evanmwillhite** [comment]
This is awesome! I upgraded the yarn lock, so that should help with yarn issues. Also, on emulsify-gulp#stylelint, I set the 'failAfterError' option to false in `cssCompile`, so that it just notifies in terminal w/o failing completely. 

I'm good with trying the stylelint-config-standard rules but my only question is - do we want to keep the stylelint config local in Emulsify rather than in its own node_module so that other users can more easily tweak as they see fit? This is pretty opinionated. I think I'd personally rather do this, but I'm interested to hear from everybody else.

**Pull request merged**